locust/util/time.py

```diff
@@ -0,0 +1,25 @@
+import re
+from datetime import timedelta
+
+def parse_timespan(time_str):
+    """
+    Parse a string representing a time span and return the number of seconds.
+    Valid formats are: 20, 20s, 3m, 2h, 1h20m, 3h30m10s, etc.
+    """
+    if not time_str:
+        raise ValueError("Invalid time span format")
+    
+    if re.match(r'^\d+$', time_str):
+        # if an int is specified we assume they want seconds
+        return int(time_str)
+    
+    timespan_regex = re.compile(r'((?P<hours>\d+?)h)?((?P<minutes>\d+?)m)?((?P<seconds>\d+?)s)?')
+    parts = timespan_regex.match(time_str)
+    if not parts:
+        raise ValueError("Invalid time span format")
```

Comment:
small nit: it'd be nice if this message showed the valid or expected formats from the docstring

Reply:
Fixed!